### PR TITLE
Add option to strip path prefix when uploading files.

### PR DESCRIPTION
**Pull request by astroilov**
See issue: https://issues.jenkins-ci.org/browse/JENKINS-23670

### CODE REVIEW

File: src/main/java/com/google/jenkins/plugins/storage/AbstractBucketLifecycleManager.java [hunk 1]
```diff
@@ -62,13 +62,14 @@
...
-   * base, which are mainly relevant for true uploads.
+   * NOTE: this hides the "make public", "for failed" and "strip path prefix"
+   * options of the base, which are mainly relevant for true uploads.
    */
   public AbstractBucketLifecycleManager(String bucket,
       @Nullable UploadModule module) {
     super(bucket, false /*sharedPublicly*/,
-        false /*forFailedJobs*/, module);
+        false /*forFailedJobs*/, false /*stripPathPrefix */,
+        null /*pathPrefix */, module);
```
Comment (tcnghia) [inline src/main/java/com/google/jenkins/plugins/storage/AbstractBucketLifecycleManager.java]: Can we combine 'pathPrefix' and 'stripPathPrefix' and only strip when 'pathPrefix' isn't null?

File: src/main/resources/com/google/jenkins/plugins/storage/AbstractUpload/help-pathPrefix.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<div>
+  <p>
+    The specified prefix will be stripped from all uploaded filenames.
```
Comment (tcnghia) [inline src/main/resources/com/google/jenkins/plugins/storage/AbstractUpload/help-pathPrefix.html]: Add description about non-match case?

**Comment by tcnghia** [comment]
Hi, @astroilov I've added some comments.

**Comment by astroilov** [comment]
Comments addressed in latest commit.

**Comment by tcnghia** [comment]
LGTM, feel free to merge

**Pull request merged**